### ISSUE TITLE
rbind.hyperframe crashes within loop after update

### ISSUE THREAD

**Issue by mvogtland** [spatstat/spatstat#131]
As already mentioned in [stackoverflow](https://stackoverflow.com/questions/61488641/spatstat-rbind-hyperframe-crashes-after-update-to-v1-63), I have a problem with rbind.hyperframe within a loop. I found out by comparing with another computer (spatstat v1.60), that rbind.hyperframe now creates row names. Those cause the crash. I checked it by creating a self-written function without setting row names. The self-written function is also similar to rbind.hyperframe in v1.60.

``` r
library(spatstat)
#> Lade nötiges Paket: spatstat.data
#> Lade nötiges Paket: nlme
#> Lade nötiges Paket: rpart
#> 
#> spatstat 1.63-3       (nickname: 'Wet paint') 
#> For an introduction to spatstat, type 'beginner'
h <- hyperframe(1)
hh <- h
for (i in 1:3) {
  hh <- rbind.hyperframe(hh, h)
}
#> Error in (function (..., row.names = NULL, check.rows = FALSE, check.names = TRUE, : angegebene Zeilennamen haben falsche Länge
print(h)
#> Hyperframe:
#>   V1
#> 1  1
print(hh)
#> Hyperframe:
#>      V1
#> X1    1
#> X1.1  1


RbindHyperframe <- function(...) {
  argh <- list(...)
  if (length(argh) == 0) {
    return(NULL)
  }
  argh <- lapply(argh, as.hyperframe)
  nargh <- length(argh)
  if (nargh == 1) {
    return(argh[[1L]])
  }
  dfs <- lapply(argh, as.data.frame, discard = FALSE)
  dfall <- do.call(rbind, dfs)
  dfs0 <- lapply(argh, as.data.frame, discard = TRUE, warn = FALSE)
  df0all <- do.call(rbind, dfs0)
  rslt <- list()
  nam <- names(dfall)
  nam0 <- names(df0all)
  for (k in seq_along(nam)) {
    nama <- nam[k]
    if (nama %in% nam0) {
      rslt[[k]] <- dfall[, k]
    }
    else {
      hdata <- lapply(argh, "[", j = nama, drop = FALSE)
      hdata <- lapply(lapply(hdata, as.list), getElement,
        name = nama
      )
      hh <- hdata[[1L]]
      for (j in 2:nargh) {
        hh <- append(hh, hdata[[j]])
      }
      rslt[[k]] <- hh
    }
  }
  # rona <- sapply(dfs, row.names)
  # rona <- make.names(rona, unique = TRUE)
  names(rslt) <- nam
  out <- do.call(hyperframe, append(rslt, list(stringsAsFactors = FALSE))) # removed: , row.names = rona
  return(out)
}

h <- hyperframe(1)
hh <- h
for (i in 1:3) {
  hh <- RbindHyperframe(hh, h)
}
print(h)
#> Hyperframe:
#>   V1
#> 1  1
print(hh)
#> Hyperframe:
#>   V1
#> 1  1
#> 2  1
#> 3  1
#> 4  1
```

<sup>Created on 2020-04-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
sessionInfo()
#> R version 3.6.3 (2020-02-29)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 17134)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
#> [3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
#> [5] LC_TIME=German_Germany.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] spatstat_1.63-3     rpart_4.1-15        nlme_3.1-144       
#> [4] spatstat.data_1.4-3
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.3            knitr_1.28            tensor_1.5           
#>  [4] magrittr_1.5          splines_3.6.3         spatstat.utils_1.17-0
#>  [7] lattice_0.20-38       rlang_0.4.5           stringr_1.4.0        
#> [10] highr_0.8             tools_3.6.3           grid_3.6.3           
#> [13] mgcv_1.8-31           xfun_0.12             deldir_0.1-25        
#> [16] htmltools_0.4.0       yaml_2.2.1            abind_1.4-5          
#> [19] goftest_1.2-2         digest_0.6.25         Matrix_1.2-18        
#> [22] evaluate_0.14         rmarkdown_2.1         polyclip_1.10-0      
#> [25] stringi_1.4.6         compiler_3.6.3
```

</details>

**Comment by baddstats** [comment]
This is a bug and will be fixed shortly. 
Thanks for notifying us.

**Comment by baddstats** [comment]
This is now fixed, in `spatstat 1.64-0.001`
